src/agent/coverage/src/cobertura.rs

```diff
@@ -32,30 +90,43 @@ pub fn cobertura(source_coverage: SourceCoverage) -> Result<String, Error> {
     )?;
 
     emitter.write(XmlEvent::start_element("packages"))?;
-    emitter.write(
-        XmlEvent::start_element("package")
-            .attr("name", "0")
-            .attr("line-rate", "0")
-            .attr("branch-rate", "0")
-            .attr("complexity", "0"),
-    )?;
+    // path (excluding file name) will be package name for better results with ReportGenerator
+    // class name will be full file path and name
+    for file in &source_coverage.files {
+        let package_line_values = compute_line_values_package(file);
+        let path = Path::new(&file.file);
+        let _file_check = path.file_name();
+        let none_message = "Invalid file format: ".to_owned() + &file.file;
+        let parent_path = match _file_check {
```

Comment:
This is all great! I suggest factoring the logic that computes `parent_path` (or its falllback) into its own function.

Reply:
Done!